src/wasm/extract.hpp

```diff
@@ -88,6 +89,7 @@ class extractor {
 
   static extractor* singleton_instance;
   static std::once_flag init_instance_flag;
+  volatile bool aborted;
```

Comment:
I don't really think `volatile` is necessary here. The compiler sees all of the needed usages, so it won't do any unwanted stuff. The only thing I would think about is if that bool could be changed from other thread. If so, maybe introducing any sync capabilities would do.